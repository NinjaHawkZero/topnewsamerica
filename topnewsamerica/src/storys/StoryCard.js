import React, {useContext, useState} from "react";
import {Card, Button} from "antd";
import "./StoryCard.css";
import UserContext from "../auth/UserContext";

const {Meta} = Card;

//Shows information about a story




function StoryCard({author, title, description, url,  urlToImage, publishedAt}) {
    let story = {author, title, description, publishedAt, url, urlToImage};
    const {saveStory} = useContext(UserContext);
    const [saved, setSaved] = useState();


    //Saves Story for User
    async function handleSave(evt){
        saveStory(story);
        setSaved(true);
    }

    return (
      <div className="Story">
       
      <Card
      key={url}
      hoverable
      style={{ width: "50%" }}
      cover={<img alt="image" src={urlToImage} className="img" />}
    >
      <Meta title={title} description={description} publishedAt={publishedAt} author={author} />
      <a href={url} target="_blank" rel="noopener noreferrer">
        <Button type="primary" style={{ marginTop: "10px" }} onClick={handleSave} disabled={saved}>
        {saved ? "Saved!" : "View & Save"}
        </Button>
      </a>
    </Card>
    </div>
    )
}

export default StoryCard;

//{urlToImage && <img src={urlToImage} className="float-right ml-5"  className="card-img-top"/>}




/*
<div className="card w-50" > {saved}


{urlToImage && <img src={urlToImage} className="float-right ml-5"  className="card-img-top"/>}
<div class="card-header">{title}</div>
 <div className="card-body">

   <h6 className="card-title">{author}</h6>
   
   <p class="card-text">{url}</p>
   <p class="card-text">{publishedAt}</p>
    <p class="card-text">{description}</p>
    
 
    <button
       className="btn btn-danger font-weight-bold text-uppercase float-right"
     
       onClick={handleSave}
       disabled={saved}
   >
     {saved ? "Saved!" : "Save"}
   </button>

 </div>

</div>

.card {
  
  max-height: 250px;
  margin-bottom: 1rem;
  color: inherit;
  border-radius: 0;
  border: none;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);

}

  .card-img-top {
    width: 50px;
    height: 50px;
    object-fit: cover;
}


*/