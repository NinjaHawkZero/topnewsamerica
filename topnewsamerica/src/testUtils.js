import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "testpass",
  
};

const save = async function saveStory(story) {
  try{let res = await TopNewsApi.saveStory(story, currentUser.username);

  return {success: true}}
  catch(err) {
    console.error("Could not save story", err);
    return {success: false, err}
  }
}

const UserProvider =
    ({ currentUser = demoUser, saveStory = save}) => (
    <UserContext.Provider value={{ currentUser, saveStory }}>
      
    </UserContext.Provider>
);

export { UserProvider };