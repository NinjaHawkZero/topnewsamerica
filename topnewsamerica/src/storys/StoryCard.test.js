import React from "react";
import { render } from "@testing-library/react";
import StoryCard from "./StoryCard";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router-dom";




it("matches snapshot with a story", function () {
    let story = { author:"Jon Wayne", title:"War in the middleeast", description:"There are conflicting forces in the middleeast", published_at:"6:45pm", url:"https://www.macrumors.com/2022/01/01/apple-watch-life-saving-911-ad/", urlToImage:"https://images.macrumors.com/t/PvHWwlMR5LrYla3IjTpwcep860o=/1600x/article-new/2022/01/apple-watch-911-ad.jpeg"};
    const { asFragment } = render(
        <MemoryRouter>
      <UserProvider>
             <StoryCard story={story} />
      </UserProvider>
      </MemoryRouter>
        
    );
    expect(asFragment()).toMatchSnapshot();
  });