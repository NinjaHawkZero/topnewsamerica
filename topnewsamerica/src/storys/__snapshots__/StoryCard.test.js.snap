// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with a story 1`] = `<DocumentFragment />`;
